### ISSUE TITLE
[BUG] FreeRSS sometimes forgets feeds that I already read

### ISSUE THREAD

**Issue by luca76** [FreshRSS/FreshRSS#6048]
**Describe the bug**
[BUG] FreeRSS sometimes forgets feeds that I already read

**To Reproduce**
1. use this feed https://www.filfre.net/feed/
2. wait some day (from 4 days to 1 week)
3. you see feeds that you alread read

**note: since I read all those feeds in another RSS read, I used "Mark all read". Maybe this is the source of problems? This happens  with other RSS feeds, too.**

**Comment by luca76** [comment]
I am ussing the FreeRSS version in the https://feeds.flossboxin.org.in/ website. They state that the version is 1.23.2-dev

**Comment by ahnlak** [comment]
I'm running 1.23.1 (in a shared hosting environment, but I'm not sure that matters) and I'm finding the same. 

It's not all feeds, but for example https://www.raspberrypi.com/feed/ fairly consistently re-appears as full of unread every day or two - looking at the feed, the update date of individual items don't appear to have changed, and I wasn't having this problem with an earlier version (1.20.2 - I hadn't updated for a while...)

I'd be happy to turn up any logging options, but I haven't come across any useful logs to track down what's happening yet...

**Comment by Alkarex** [comment]
When you say re-appear as unread, is it a new copy, or the same article, which state has changed back to unread?

First, make sure that you have turned off the option to automatically mark as unread the articles that have changed.

Then, an helpful way to debug those cases is to download and keep different copies of the feed, and when a problem is observed, post here the same article in a copy from before the problem and from after (I hope that makes sense, otherwise I can try to reformulate).
Typically, a GUID or something else has changed.

**Comment by ahnlak** [comment]
I've made sure that option is turned off, and saved a copy of the feed as it is now; hopefully it will go wrong again in a couple of days and we can compare then!

The interesting thing is that this never occurred on 1.20.2, and has happened to multiple feeds since moving to 1.23.1 (although I'm aware that I also switched from VPS to shared hosting, so maybe it's something permissions related - although I would expect that to affect *all* feeds...)

**Comment by luca76** [comment]
@Alkarex the problem is that, to me, this is happening on all my RSS feeds. Some from blogspot, some for others (like the link in the first post)

**Comment by luca76** [comment]
@Alkarex  when you said: 

> First, make sure that you have turned off the option to automatically mark as unread the articles that have changed.

I cannot find the option, can you put me a screenshot ?

**Comment by luca76** [comment]
See here, it's happening also from official FreeRSS RSS feed...

![image](https://github.com/FreshRSS/FreshRSS/assets/5095918/14ba9047-0585-4164-9e2a-0d50914d270f)

**Comment by Alkarex** [comment]
> See here, it's happening also from official FreeRSS RSS feed...

This looks completely fine?

**Comment by Alkarex** [comment]
> I cannot find the option, can you put me a screenshot ?

@luca76 Under configure > reading:
![image](https://github.com/FreshRSS/FreshRSS/assets/1008324/37477e7f-4054-49de-830a-872e69852f65)

**Comment by luca76** [comment]
> > See here, it's happening also from official FreeRSS RSS feed...
> 
> This looks completely fine?

NO. It's 4th time that same feeds from FreeRSS RSS are reappearing as unread.

**Comment by luca76** [comment]
> > I cannot find the option, can you put me a screenshot ?
> 
> @luca76 Under configure > reading: 

Thank you. It's already turned off for me,

**Comment by Alkarex** [comment]
> NO. It's 4th time that same feeds from FreeRSS RSS are reappearing as unread.

Hum, I register to https://github.com/FreshRSS/FreshRSS/releases.atom from multiple FreshRSS instances, and have not observed this problem.

**Comment by luca76** [comment]
> Hum, I register to https://github.com/FreshRSS/FreshRSS/releases.atom from multiple FreshRSS instances, and have not observed this problem.

It's my same URL address. Maybe can you try at https://feeds.flossboxin.org.in/ ?

**Comment by luca76** [comment]
> See here, it's happening also from official FreeRSS RSS feed...
> 
> ![image](https://private-user-images.githubusercontent.com/5095918/298841442-14ba9047-0585-4164-9e2a-0d50914d270f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYwODczMDgsIm5iZiI6MTcwNjA4NzAwOCwicGF0aCI6Ii81MDk1OTE4LzI5ODg0MTQ0Mi0xNGJhOTA0Ny0wNTg1LTQxNjQtOWUyYS0wZDUwOTE0ZDI3MGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMjRUMDkwMzI4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTFhODIxNDM0MGUwNzJkNWNiNjhmN2E0OTYyMjFjMjFkYjY5MTU4NzEwMGQ5NzcwNmMxZDBmMWM0MTg1MGQ0OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.-_Sh809Rt_WRuXnav8EPihIbHYFySDRI0GeW19PBKtc)

It did this again today...

**Comment by Alkarex** [comment]
Regarding flossboxin:
* https://github.com/FreshRSS/FreshRSS/issues/5568
* https://github.com/FreshRSS/FreshRSS/issues/6032

Maybe you could reach out to the admin so that we can do some debugging. Maybe some other things are going on on that instance.

**Comment by ahnlak** [comment]
I'm starting to suspect my issues were different; it only happened to a few feeds and having deleted / re-added them it seems to have stopped happening - was a fresh install with an import from a different instance, so maybe there were some sync issues that have resolved themselves.

**Comment by MarcusWolschon** [comment]
I have 1.24.2-dev and "mark updated as unread" is not set.
This happens every now and then with https://www.heise.de/rss/heise-atom.xml
I'm starting to save a copy of that feed daily for a while to check if something is wrong with the feed.

**Comment by Frenzie** [comment]
Also see https://github.com/FreshRSS/FreshRSS/issues/5151 in case there's WebSub involved.

**Comment by Alkarex** [comment]
@MarcusWolschon Debugging welcome, and please check https://github.com/FreshRSS/FreshRSS/discussions/6101

**Comment by yuliu** [comment]
> See here, it's happening also from official FreeRSS RSS feed...
> 
> ![image](https://github.com/FreshRSS/FreshRSS/assets/5095918/14ba9047-0585-4164-9e2a-0d50914d270f)

Some feeds for GitHub release pages that contain images hosted by GitHub have issues for me, like the OP's. Among these images, some have the URLs beginning with `https://private-user-images.githubusercontent.com/...` and these URLs would usually change across different visits. Therefore, feeds that contain those image URLs would also change upon feed updating, although the contents of these feeds are visually unchanged.

Relevant feeds having issues mentioned above at the time of writing, for test:
 - https://github.com/photoprism/photoprism/releases.atom
 - https://github.com/rustdesk/rustdesk/releases.atom
 - https://github.com/th-ch/youtube-music/releases.atom
